### PR TITLE
python312Packages.gradio: 4.44.1 -> 5.1.0

### PR DESCRIPTION
**Pull request by pbsds**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by GaetanLepage — APPROVED** [review]
LGTM !
If you have the patience, maybe go with:
- getting rid of `with lib;` in the `meta` attrs
- `nativeBuildInputs` -> `build-system` in `gradio`

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349625`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/gradio-pdf/default.nix [hunk 1]
```diff
@@ -11,16 +11,16 @@
 
 buildPythonPackage rec {
   pname = "gradio-pdf";
-  version = "0.0.15";
+  version = "0.0.17";
   format = "pyproject";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/gradio-pdf/default.nix:15]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/gradio-pdf/default.nix [hunk 2]
```diff
@@ -11,16 +11,16 @@
...
 
   src = fetchPypi {
     pname = "gradio_pdf";
     inherit version;
-    hash = "sha256-R+4CDJo84/+PUTTxoz9CsZlHUVsldgjufWsRkALrfN4=";
+    hash = "sha256-LoVcwE7eGcK5Nc6qKTnrnI+rNlsDbekhKUP+Fzq2SQ8=";
   };
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/gradio-pdf/default.nix:21]: Have you tried using the github repo instead ?

File: pkgs/development/python-modules/gradio/default.nix [hunk 3]
```diff
@@ -64,15 +64,15 @@
 
 buildPythonPackage rec {
   pname = "gradio";
-  version = "4.44.1";
+  version = "5.1.0";
   format = "pyproject";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/gradio/default.nix:68]:
```suggestion
  pyproject = true;
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/gradio-pdf/default.nix (hunk 2 above)
Comment (pbsds) [reply pkgs/development/python-modules/gradio-pdf/default.nix:21]: no tags

**Review by GaetanLepage — APPROVED** [review]
LGTM !

File: pkgs/development/python-modules/gradio-pdf/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/gradio-pdf/default.nix:21]: Ok :'(, thanks for trying.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349625`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Failures are unrelated. Feel free to merge :)

**Comment by pbsds** [comment]
will have to wait for stable branch-off

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349625`

---
### `x86_64-linux`
<details>
  <summary>:x: 10 packages failed to build:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349625`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gradio-client.dist</li>
    <li>python311Packages.gradio-pdf</li>
    <li>python311Packages.gradio-pdf.dist</li>
    <li>python311Packages.gradio.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]
`private-gpt` failures are surely unrelated.

**Pull request merged**